### PR TITLE
Give proper AI comms to cyborg shells

### PR DESCRIPTION
**Pull request by Arahimine**
## About the PR 
[feat]
This PR enables cyborg shells controlled by the AI to use all the same frequencies as the AI itself. Previously, they could only use General, machine talk and the frequency related to their module.

Similar fix for regular AI shells coming soon. I tried to do it first but then went insane.

## Why's this needed?
The AI's whole job description is to monitor comms constantly, and deploying into a shell (as it often has to do) cripples that. It's so bad that any AI worth their salt publicly broadcasts that they're deploying to shell just so they don't miss any messages. The only counterargument I can conceive of is that an AI being off-comms allows antags and crimers to get away with things, but if their misdeeds are being broadcast over comms anyway, chances are they're already screwed, AI or not.

## Changelog
```
(u)Arahimine:
(+)AI-controlled cyborg shells now have access to the same communications channels as the AI.
```

### CODE REVIEW

**Review by mordent-goonstation — COMMENTED** [review]
Not commenting on code directly as it's an overall thing: could you verify that behavior is expected if the cyborg's brain is switched out to/from an AI interface board and a regular person's brain, with regards to what radios each can communicate on. Same, but vice-versa.

I'd need to go read the radio code in a little more detail to understand your implementation, but my initial read implies this might not be covered.

**Comment by Arahimine** [comment]
Yeah, my code's a little icky because I'm a novice, it's 2 AM and I recently downed an entire bottle of cider.
But for what I think you're asking (again, tired and a bit drunk), it does account for switching an interface board into an already-initialized borg and vice versa. Or at least it's supposed to, I don't have other people to test it with, but the code for it has no reason not to work and it does compile, so. If you see any problems that I missed you're more than welcome to tell me where I fucked up, of course.

File: code/mob/living/silicon/robot.dm [hunk 1]
```diff
@@ -1824,7 +1837,10 @@
 		hud.update_module()
 		hud.module_added()
 		if(istype(RM.radio))
-			src.radio = RM.radio
+			if (src.shell)
+				src.radio = src.ai_radio
+			if (!src.shell)
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:1842]: else

File: code/mob/living/silicon/robot.dm [hunk 2]
```diff
@@ -1839,7 +1855,10 @@
 		hud.module_removed()
 		if(istype(src.radio) && src.radio != src.default_radio)
 			src.radio.set_loc(RM)
-			src.radio = src.default_radio
+			if (!src.shell)
+				src.radio = src.default_radio
+			if (src.shell)
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:1860]: switch and else

File: code/mob/living/silicon/robot.dm [hunk 3]
```diff
@@ -184,7 +185,11 @@
 
 			src.botcard.access = get_all_accesses()
 			src.default_radio = new /obj/item/device/radio(src)
-			src.radio = src.default_radio
+			src.ai_radio = new /obj/item/device/radio/headset/command/ai(src)
+			if (!src.shell)
+				src.radio = src.default_radio
+			if (src.shell)
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:191]: flip and else

**Review by ZeWaka — COMMENTED** [review]
don't know why you made it so I can't make suggestions....

**Comment by ZeWaka** [comment]
You deleted your repository or some weird shit so i'ma remake this.

**Pull request closed without merging**